hibernate-core/src/main/java/org/hibernate/query/sqm/sql/internal/EntityValuedPathInterpretation.java

```diff
@@ -357,31 +359,17 @@ private static boolean isSelected(TableGroup tableGroup, NavigablePath path, Sqm
 	}
 
 	private static boolean selectClauseContains(NavigablePath path, SqmQuerySpec<?> sqmQuerySpec) {
-		final List<SqmSelection<?>> selections = sqmQuerySpec.getSelectClause() == null
+		final List<SqmSelectableNode<?>> selections = sqmQuerySpec.getSelectClause() == null
 				? Collections.emptyList()
-				: sqmQuerySpec.getSelectClause().getSelections();
-		for ( SqmSelection<?> selection : selections ) {
-			if ( selectableNodeContains( selection.getSelectableNode(), path ) ) {
+				: sqmQuerySpec.getSelectClause().getSelectionItems();
```

Comment:
`SqmSelectClause#getSelectionItems` is unnecessarily materializing a new list, so it would be preferrable if you could stick to using the original approach. I suppose you'd have to add another else-if block for compound selections.

Reply:
I thought using that method would be the safest bet, in case I forgot to handle any other specific case.

I've now gone back to the previous implementation as suggested, adding a case for compound selections.